zooniverse/front-end-monorepo
# Bump @testing-library/dom and fix ZooHeader console warnings

## Package
lib-react-components

## Linked Issue and/or Talk Post
Includes https://github.com/zooniverse/front-end-monorepo/pull/4395

## Describe your changes

- This PR branches off of the linked dependabot PR https://github.com/zooniverse/front-end-monorepo/pull/4395
- I added `waitFor` in ZooHeader's unit tests when checking for opened menu lists
- ZooHeader used to throw a lot of console warnings in dev and test environments because of incorrectly expecting a `user` prop while signed out

## How to Review

- Check ZooHeader's storybook for various rendered states.
- Visit i-fancy-cats while running this branch locally. Test that you can sign in and and sign out on both desktop (wide) screens and mobile (narrow) screens.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook


## Bug Fix

## Maintenance
- [ ] If not from dependabot, the PR creator has described the update (major, minor, or patch version, changelog)
